pkg/providers/tinkerbell/controlplane.go

```diff
@@ -51,6 +55,23 @@ func ControlPlaneSpec(ctx context.Context, logger logr.Logger, client kubernetes
 		return nil, errors.Wrap(err, "generating tinkerbell control plane yaml spec")
 	}
 
+	capiCluster, err := controller.GetCAPICluster(ctx, client, clusterSpec.Cluster)
+	if err != nil {
+		return nil, err
+	}
+	if capiCluster != nil {
+		rollingOverScale, err := anywherev1.IsRollingNotScale(ctx, client, clusterSpec.Cluster)
+		if err != nil {
+			return nil, errors.Wrap(err, "checking upgrade behavior")
+		}
+		if rollingOverScale {
+			controlPlaneYaml, err = omitTinkerbellMachineTemplate(controlPlaneYaml)
```

Comment:
This maps to [current CLI logic ](https://github.com/aws/eks-anywhere/blob/75eb25f76ffc39344302098cdaddce463efcfea7/pkg/providers/tinkerbell/template.go#L292
)

Reply:
I see. However I don't think that's a good reason to do it this way. The CLI logic operates over yaml because it was never able to operate over api structs. This code that you are adding here is basically adding that capability.